functionary/ui/templates/partials/modal.html

```diff
@@ -0,0 +1,26 @@
+<div class="modal is-active" id="form-modal">
+    <div class="modal-background"></div>
+    <div class="modal-card">
+        <header class="modal-card-head">
+            <h1 class="modal-card-title is-2">
+                {% block modal_title %}{% endblock %}
+            </h1>
+            <button class="delete"
+                    aria-label="close"
+                    onclick="return this.parentNode.parentNode.parentNode.remove();"></button>
+        </header>
+        {% block form_declaration %}{% endblock %}
```

Comment:
Yea, I couldn't think of another way to structure the modal template so that the forms wouldn't be too highly coupled with the modal. This primarily stems from wanting to use that modal footer section for the buttons. This problem might go away if we switch to daisyUI, where their modal layout does not include a footer section, so this implementation would look very similar to my original implementation of the modal template.